src/services/socket.ts

```diff
@@ -0,0 +1,47 @@
+import { Server } from 'socket.io';
+import http from 'http';
+import { socketConfig, messageConfig } from '../config/socket';
+
+export const socketWorker = (
+  server: http.Server<typeof http.IncomingMessage, typeof http.ServerResponse>
+) => {
+  const io = new Server(server, {
+    path: '/socket.io',
+    cors: {
+      origin: '*', // this is for local testing
+    },
+  });
+
+  io.on('connection', socket => {
+    console.log('a user connected : ', socket.id);
+    socket.on('join', (data: string) => {
+      const requestData = JSON.parse(data);
+      console.log(requestData, requestData['clientId']);
+      socketConfig[requestData['clientId']] = socket.id;
+      console.log(socketConfig);
+    });
+    socket.on('disconnect', () => {
+      const key = Object.keys(socketConfig).find(key => socketConfig[key] === socket.id);
+      if (key) delete socketConfig[key];
+      console.log('user disconnected');
+    });
+    socket.on('get_status', (data: string) => {
```

Comment:
get status should be associated with `apiKey` or `clientId` as message ids might not be persisted on client or get renewed on failure.